dtao.c

```diff
@@ -345,13 +398,138 @@ static struct zwlr_layer_surface_v1_listener layer_surface_listener = {
 	.closed = layer_surface_closed,
 };
 
+struct input_state {
+	struct wl_surface *surface;
+	double x, y;
+	uint32_t button;
+};
+
+static void
+noop() {}
+
+static void
+wl_pointer_enter(void *data, struct wl_pointer *wl_pointer,
+			uint32_t serial, struct wl_surface *surface,
+			wl_fixed_t surface_x, wl_fixed_t surface_y)
+{
+	struct input_state *istate = data;
+	istate->surface = surface;
+	istate->x = wl_fixed_to_double(surface_x);
+	istate->y = wl_fixed_to_double(surface_y);
+}
```

Comment:
I re-read the protocol, and it says:
```
When a seat's focus enters a surface, the pointer image
is undefined and a client should respond to this event by setting
an appropriate pointer image with the set_cursor request.
```

Reply:
I think that's better suited for a separate pr